src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CspKeyContainerInfo.cs

```diff
@@ -238,15 +188,64 @@ public string UniqueKeyContainerName
         {
             get
             {
-                SafeProvHandle safeProvHandle = SafeProvHandle.InvalidHandle;
-                int hr = CapiHelper.OpenCSP(_parameters, (uint)CapiHelper.CryptAcquireContextFlags.CRYPT_SILENT, ref safeProvHandle);
+                return (string)ReadKeyParameterSilent(Constants.CLR_UNIQUE_CONTAINER);
+            }
+        }
+
+        /// <summary>
+        /// Read a the parameter from the current key using CRYPT_SILENT, to avoid any potential UI prompts.
+        /// </summary>
+        private object ReadKeyParameterSilent(int keyParam, bool throwOnNotFound=true)
+        {
+            const uint SilentFlags = (uint)CapiHelper.CryptAcquireContextFlags.CRYPT_SILENT;
+
+            SafeProvHandle safeProvHandle;
+            int hr = CapiHelper.OpenCSP(_parameters, SilentFlags, out safeProvHandle);
```

Comment:
Some of the methods you replaced used the verify context. Do we need that for something?